add blog

il3ven opened the pull request:
This PR adds blog functionality using 11ty and EleventyDuo theme. Live demo at https://il3ven.github.io/neume-network-website. Fixes #1.

### Changes:

#### Fix for GitHub Page
The theme is now deployable to Github pages. Github hosts pages at `user.github.io/repo` so links break. For example a link to `/posts` will point to `user.github.io/posts` instead of `user.github.io/repo/posts`. This was one of the fix.

#### Comment unnecessary code
I have commented out [mermaid](https://www.npmjs.com/package/mermaid) which is a diagramming tool. This reduces bundle size by 300kb. (Lighthouse scores are now 🚀)

#### Personalise Blog
I have personalised the blog i.e mention neume-network and change social links. `src/data/site.json` contains many constants.

#### Add workflow to deploy to Github Pages
I have added a workflow to deploy to gh pages. In order for Github pages to work on neume-network/website repo there are a few steps:
- Enable Github pages from settings
- Select gh-pages as the branch and root as the directory
- Create a gh-pages manually. The workflow doesn't have permission to create new branches. It can only update.

I can do the above steps but I will need access to the repo.

### Further Changes

We can still personalise it more. Change the text on the welcome or about page. We can even add new pages or change some css styling. Let me know if any further changes are required. We can open new issues for that. Though in the current state the blog does good. Anymore changes will be subjective.

### TODO

- [x] Fix sitemap
- [x] Change font
- [x] Add readme on how to add new posts
- [x] Incorporate plausible

il3ven committed:
add gitignore

il3ven committed:
use eleventy-duo as template with minor modifications

- Switched from yarn to npm
- Updated webpack* to latest versions

il3ven committed:
update post-css-* packages to latest version

il3ven committed:
disable mermaid charts

This will improve the time to render and thus improve lighthouse score.

il3ven committed:
personalize blog

il3ven committed:
add github workflow

deploy on push to add-blog

move deploy.yml to .github/workflows

add permission in workflow

add prefix path

il3ven committed:
add path prefix and also fix missing url filter

il3ven committed:
change workflow to work on neume-network/website

The earlier changes was for development.

il3ven committed:
change title of home page

TimDaub reviewed (changes requested):
- It looks clean, and also I like that the page has a small footprint :)
- A minor legal annoyance: For the EU, Google Fonts isn't GDPR compliant by default: https://usercentrics.com/knowledge-hub/google-fonts-gdpr-compliant/
  - I recommend finding either a web-safe font similar to what we're already using or vendoring in the font as shown in the article above.
  - Alternative is including a cookie banner (which I'd like to avoid)

il3ven commented:
> A minor legal annoyance: For the EU, Google Fonts isn't GDPR compliant by default.

I'll look into this. I didn't know about Google fonts and GDPR. Which font are we currently using?

I also found that the sitemap isn't working. This [problem](https://eleventyduo.netlify.app/sitemap.xml) is present in the theme too. I'll look into this too.

TimDaub commented on src/layouts/base.njk:
```diff
@@ -0,0 +1,149 @@
+    {% bundledcss %}
+
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=IBM+Plex+Sans&display=swap" rel="stylesheet">
```
replace bc of gdpr

TimDaub commented on src/css/typography.css:
```diff
@@ -0,0 +1,60 @@
+  font-variant-ligatures: contextual;
+  -webkit-overflow-scrolling: touch;
+  -webkit-text-size-adjust: 100%;
+  font-family: 'IBM Plex Sans', sans-serif;
```
replace bc of gdpr

reimertz commented:
Thanks @TimDaub for bringing up not using Google fonts. 

On that note,I have setup a Cloudflare worker to enable a [Plausible.io](https://plausible.io) proxy for a privacy-first, GDPR-complaint cookieless analytics.

All we have to do is to add a script tag with `https://neume.network/neume/js/info.js` and everything else should work as intended.

```
<script defer data-domain="neume.network" src="https://neume.network/neume/js/info.js"></script>
```

TimDaub commented:
> On that note,I have setup a Cloudflare worker to enable a [Plausible.io](https://plausible.io) proxy for a privacy-first, GDPR-complaint cookieless analytics.

That's awesome! Thanks!

reimertz commented:
I took the liberty of connecting this repo to Netlify for smooth preview deploy integration. So hopefully, the next commit will generate a preview link. 

Once the site is deployed, I'll help setting up Cloudflare cache purge so the site can be proxied by Cloudflare CDN with aggressive cache, but then purged once new deploys happen.

Lmk if this sounds good, if not, I can revert the integration. :)

reimertz committed:
Make small change to README.md 

Made this change to trigger a netlify preview build.

reimertz commented:
Netlify deploy preview seems to be working smoothly now.

djfnd commented:
Apologies, I missed the PR previously.
Font-wise, we are running under the assumption of Helvetica for the design work that is ongoing, if that helps.

il3ven committed:
updated favicon

il3ven committed:
fix sitemap

il3ven committed:
add plausible

il3ven committed:
add a section on how to add blog post

il3ven committed:
change font to Helvetica Neue

This is done because of GDPR.

Also changed line-height to be more suitable for the new font.

il3ven committed:
few css changes to fix margin between h1, h2...

TimDaub reviewed (approved):
great job @il3ven!
